scipy/stats/stats.py

```diff
@@ -4363,10 +4363,8 @@ def ks_2samp(data1, data2):
     data2 = np.sort(data2)
     n1 = data1.shape[0]
     n2 = data2.shape[0]
-    data_all = np.concatenate([data1, data2])
-    cdf1 = np.searchsorted(data1, data_all, side='right') / (1.0*n1)
-    cdf2 = np.searchsorted(data2, data_all, side='right') / (1.0*n2)
-    d = np.max(np.absolute(cdf1 - cdf2))
+    from . import _stats
+    d = _stats.ks_2samp(data1, data2)
```

Comment:
It might actually be worth keeping the old lines and just commenting them out, saying your code implements something equivalent to those lines, but more efficiently.